File: LeetCode/Medium/406 Queue Reconstruction by Height.js
/**
 * @param {number[][]} people
 * @return {number[][]}
 */
var reconstructQueue = function (people) {
  people.sort((a, b) => b[0] - a[0] || a[1] - b[1]);

  let res = [];
  for (let [h, k] of people) {
    res.splice(k, 0, [h, k]);
  }

  return res;
};

File: LeetCode/Medium/406 Queue Reconstruction by Height.py
class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        people = sorted(people, key = lambda x : (x[0], -x[1]), reverse=True)

        q = []
        for val in people:
            q = q[:val[1]] + [val] + q[val[1]:]
        
        return q